mongodbatlas/provider.go

```diff
@@ -337,7 +302,106 @@ func providerConfigure(ctx context.Context, d *schema.ResourceData) (interface{}
 		}
 	}
 
-	return config.NewClient(ctx)
+	client, err := config.NewClient(ctx)
+	if err != nil {
+		return nil, diag.FromErr(err)
+	}
+	return client, nil
+}
+
+func setDefaultsAndValidations(d *schema.ResourceData) error {
+	mongodbgovCloud := pointy.Bool(d.Get("is_mongodbgov_cloud").(bool))
+	if *mongodbgovCloud {
+		if err := d.Set("base_url", "https://cloud.mongodbgov.com"); err != nil {
+			return err
+		}
+	} else {
+		if err := setValueFromConfigOrEnv(d, "base_url", []string{
+			"MONGODB_ATLAS_BASE_URL",
+			"MCLI_OPS_MANAGER_URL",
+		}); err != nil {
+			return err
+		}
+	}
```

Comment:
base_url should work with both gov and not gov
```suggestion
	if *mongodbgovCloud {
		if err := d.Set("base_url", "https://cloud.mongodbgov.com"); err != nil {
			return err
		}
}
		if err := setValueFromConfigOrEnv(d, "base_url", []string{
			"MONGODB_ATLAS_BASE_URL",
			"MCLI_OPS_MANAGER_URL",
		}); err != nil {
			return err
		}
	}
```

Reply:
For this case I replicated the same behaviour we have in our existing sdk v2 provider https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/mongodbatlas/provider.go#L317.
Not sure if we would want to incorporate this change within the changes of the migration.

Reply:
This change just improves readability so I would add it

Reply:
adjusted